android/src/main/java/com/adyenreactnativesdk/cse/AdyenCSEModule.kt

```diff
@@ -0,0 +1,58 @@
+package com.adyenreactnativesdk.cse
+
+import com.adyen.checkout.cse.CardEncrypter
+import com.adyen.checkout.cse.UnencryptedCard
+import com.adyen.checkout.cse.exception.EncryptionException
+import com.facebook.react.bridge.*
+
+class AdyenCSEModule(context: ReactApplicationContext?) : ReactContextBaseJavaModule(context) {
+
+    @ReactMethod
+    fun addListener(eventName: String?) { }
+
+    @ReactMethod
+    fun removeListeners(count: Int?) { }
+    
+    override fun getName() = COMPONENT_NAME
+
+    @ReactMethod
+    fun encryptCard(card: ReadableMap, publicKey: String, promise: Promise) {
+        val unencryptedCardBuilder = UnencryptedCard.Builder()
+        card.getString(NUMBER_KEY)?.let { unencryptedCardBuilder.setNumber(it) }
+        card.getString(EXPIRY_MONTH_KEY)?.let { unencryptedCardBuilder.setExpiryMonth(it) }
+        card.getString(EXPIRY_YEAR_KEY)?.let { unencryptedCardBuilder.setExpiryYear(it) }
+        card.getString(CVV_KEY)?.let { unencryptedCardBuilder.setCvc(it) }
+
+        try {
+            val encryptedCard = CardEncrypter.encryptFields(unencryptedCardBuilder.build(), publicKey)
+            val map = WritableNativeMap()
+            map.putString(NUMBER_KEY, encryptedCard.encryptedCardNumber)
+            map.putString(EXPIRY_MONTH_KEY, encryptedCard.encryptedExpiryMonth)
+            map.putString(EXPIRY_YEAR_KEY, encryptedCard.encryptedExpiryYear)
+            map.putString(CVV_KEY, encryptedCard.encryptedSecurityCode)
+            promise.resolve(map)
+        } catch (e: EncryptionException) {
+            promise.reject(ERROR_MESSAGE, e)
+        }
+    }
+
```

Comment:
The `CardEncrypter` also has the `public static String encrypt(...)` method to encrypt all data into a single string. Did you leave that out on purpose?

Reply:
I do not know how it is used on Checkout API.
AFAIK it is for "Classic" PAL integration that was deprecated